community/kernel/src/test/java/org/neo4j/kernel/impl/index/schema/FullScanNonUniqueIndexSamplerTest.java

```diff
@@ -0,0 +1,140 @@
+ */
+package org.neo4j.kernel.impl.index.schema;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.neo4j.index.internal.gbptree.GBPTree;
+import org.neo4j.index.internal.gbptree.Writer;
+import org.neo4j.io.pagecache.IOLimiter;
+import org.neo4j.kernel.api.index.IndexEntryUpdate;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.impl.api.index.sampling.IndexSamplingConfig;
+import org.neo4j.storageengine.api.schema.IndexSample;
+import org.neo4j.test.rule.PageCacheRule;
+import org.neo4j.test.rule.RandomRule;
+import org.neo4j.test.rule.TestDirectory;
+import org.neo4j.test.rule.fs.DefaultFileSystemRule;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.rules.RuleChain.outerRule;
+
+import static org.neo4j.helpers.collection.Iterators.array;
+import static org.neo4j.index.internal.gbptree.GBPTree.NO_HEADER;
+import static org.neo4j.index.internal.gbptree.GBPTree.NO_MONITOR;
+import static org.neo4j.index.internal.gbptree.RecoveryCleanupWorkCollector.IMMEDIATE;
+import static org.neo4j.test.rule.PageCacheRule.config;
+
+public class FullScanNonUniqueIndexSamplerTest
+{
+    private final DefaultFileSystemRule fs = new DefaultFileSystemRule();
+    private final TestDirectory directory = TestDirectory.testDirectory( getClass(), fs.get() );
+    private final PageCacheRule pageCacheRule = new PageCacheRule( config().withAccessChecks( true ) );
+    protected final RandomRule random = new RandomRule();
+    @Rule
+    public final RuleChain rules = outerRule( fs ).around( directory ).around( pageCacheRule ).around( random );
+
+    private final NonUniqueSchemaNumberIndexLayout layout = new NonUniqueSchemaNumberIndexLayout();
+
+    @Test
+    public void shouldIncludeAllValuesInTree() throws Exception
+    {
+        // GIVEN
+        List<Number> values = generateNumberValues();
+        buildTree( values );
+
+        // WHEN
+        IndexSample sample;
+        try ( GBPTree<NonUniqueSchemaNumberKey,NonUniqueSchemaNumberValue> gbpTree = newTree( layout ) )
+        {
+            IndexSamplingConfig samplingConfig = new IndexSamplingConfig( Config.defaults() );
+            FullScanNonUniqueIndexSampler<NonUniqueSchemaNumberKey,NonUniqueSchemaNumberValue> sampler =
+                    new FullScanNonUniqueIndexSampler<>( gbpTree, layout, samplingConfig );
+            sample = sampler.result();
+        }
+
+        // THEN
+        assertEquals( values.size(), sample.sampleSize() );
+        assertEquals( countUniqueValues( values ), sample.uniqueValues() );
+        assertEquals( values.size(), sample.indexSize() );
+    }
+
+    static int countUniqueValues( List<Number> values )
+    {
+        int count = 0;
+        Set<Double> seenValues = new HashSet<>();
```

Comment:
`values.stream().map( Number::doubleValue ).collect( Collectors.toSet() ).size()` if you really need doubleValues
otherwise just add all values to set and get size of it